pandas/tests/window/test_expanding.py

```diff
@@ -264,3 +264,25 @@ def test_expanding_skew_kurt_numerical_stability(method):
     s = s + 5000
     result = getattr(s.expanding(3), method)()
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("window", [1, 3, 10, 50, 1000])
+@pytest.mark.parametrize("method", ["min", "max", "average"])
+@pytest.mark.parametrize("pct", [True, False])
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("test_data", ["default", "duplicates", "nans"])
+def test_rank(window, method, pct, ascending, test_data):
+    length = 1000
```

Comment:
Nit: Could we test with a smaller sample (e.g. 20) and adjust `window` accordingly?